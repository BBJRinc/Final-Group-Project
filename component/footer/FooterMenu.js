import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { Container, Header, Content, Footer, FooterTab, Button, Icon, Text } from 'native-base';

export default class FooterMenu extends Component{
    render(){
        return(
                <Footer>
                    <FooterTab>
                        <Button vertical>
                            <Icon name="add" />
                        </Button>
                        <Button vertical >
                            <Icon name="calendar" onPress={() => this.props.openDrawer('footer')} />
                            <Text>Calendar</Text>
                        </Button>
                        <Button vertical onPress={() => this.props.showMenuItem('showModal')}>
                            <Icon name="clipboard" />
                            <Text>Tasks</Text>
                        </Button>
                        <Button vertical onPress={() => this.props.showMenuItem('showModal')} >
                            <Icon name="paper" />
                            <Text>WIP</Text>
                        </Button>
                        <Button vertical>
                            <Icon name="navigate" />
                            <Text>Menu</Text>
                        </Button>
                    </FooterTab>
                </Footer>
        )
    }
}

