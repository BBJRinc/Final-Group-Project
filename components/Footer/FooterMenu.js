import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Container, Header, Content, Footer, FooterTab, Button } from 'native-base';
import IconF from 'react-native-vector-icons/Feather';


export default class FooterMenu extends Component{
    render(){
        const { unschedCount } = this.props;  
        return(
                <Footer >
                    <FooterTab light style={{backgroundColor: 'rgba(0, 0, 0, .85)'}}>
                        <Button vertical onPress={() => this.props.logout()}>
                            <IconF name="plus" size={35} color={'#fff'}/>
                        </Button>
                        <Button  vertical > 
                            <IconF name="calendar" size={35} color={'#fff'} onPress={() => this.props.showMenuItem('showCalendar')} />
                        </Button>
                        
                        <Button  vertical onPress={() => this.props.showMenuItem('showTasks')}>
                            <IconF name="check-square" size={35} color={'#fff'} />
                            {!unschedCount ? 
                            null 
                            :
                            <View style={styles.badge}>
                                <Text style={{color: '#fff'}}>{unschedCount}</Text>
                            </View>
                             }
                            
                        </Button>
                        <Button vertical onPress={() => this.props.showMenuItem('showOngoing')} >
                            <IconF name="folder" size={35} color={'#fff'} />
                        </Button>
                        <Button vertical>
                            <IconF name="menu" size={35} color={'#fff'}/>
                        </Button>
                    </FooterTab>
                </Footer>
        )
    }
}

const styles = StyleSheet.create({
    badge: {
        position: 'absolute',
        top: 1,
        right:4,
        width: 25,
        height: 25,
        borderRadius: 15,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#ff0000'
    }
})

