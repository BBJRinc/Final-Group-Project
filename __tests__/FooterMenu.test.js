//Jordan's test

import React from 'react';
import FooterMenu from '../components/Footer/FooterMenu';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<FooterMenu />).toJSON();
    expect(tree).toMatchSnapshot();
});