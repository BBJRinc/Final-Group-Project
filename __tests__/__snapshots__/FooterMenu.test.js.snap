// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "borderColor": "#cbcbcb",
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexDirection": "row",
      "height": 55,
      "justifyContent": "center",
      "left": 0,
      "paddingBottom": 0,
      "right": 0,
    }
  }
  virtual={undefined}
>
  <View
    light={true}
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, .85)",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 35,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 35,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 35,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 35,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 35,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
